src/fondant/executor.py

```diff
@@ -424,3 +425,24 @@ def _write_data(self, dataframe: dd.DataFrame, *, manifest: Manifest):
 
     def upload_manifest(self, manifest: Manifest, save_path: t.Union[str, Path]):
         pass
+
+
+COMPONENT_EXECUTOR_MAPPING: t.Dict[str, t.Type[Executor]] = {
+    "DaskLoadComponent": DaskLoadExecutor,
+    "DaskTransformComponent": DaskTransformExecutor,
+    "DaskWriteComponent": DaskWriteExecutor,
+    "PandasTransformComponent": PandasTransformExecutor,
+}
+
+
+class ComponentRunner:
```

Comment:
I would rename it to `ExecutorFactory`. In this case, the `executor.py` is the correct location where I would expect such a class.